### PR TITLE
O(N^2) O(1) sollution

### PR DESCRIPTION
**Pull request by Glazomer**
## Problem
Given a string s, return the longest palindromic substring in s.

## Approach
I have kind of brute force algorithm. I take every index, and try to find palindrome by expanding in each way.

#### O(N^2) O(1)
So, for every index N we search for palindrome for size of not more than N, so worst case complexity is O(N^2) for running and O(1) for memory, because we don't use extra memory

### CODE REVIEW

File: LongestPalindromicSubstring.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Solution:
+    def longestPalindrome(self, s: str) -> str:
+        L = R = l = r = 0
```
Comment (vitkarpov) [inline LongestPalindromicSubstring.py]: Naming is hard 😢 Wouldn't be better to name L, R as `resultLeft, resultRight`?

File: LongestPalindromicSubstring.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+        while r < len(s):
+            if l == r:
+                # Looking for the 'root' of palindrome,
+                # either odd or even number of same characters
+                while r + 1 < len(s) and s[l] == s[r+1]:
+                    r += 1
+                    if l > 0 and s[l - 1] == s[r]:
+                        l -= 1
```
Comment (vitkarpov) [inline LongestPalindromicSubstring.py:12]: Wouldn't it be better to stip out to a function like `findPalindrome`?

File: LongestPalindromicSubstring.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+            if 0 <= l and s[l] == s[r]:
+                L, R = (l, r+1) if ((r+1)-l > R-L) else (L, R)
```
Comment (vitkarpov) [inline LongestPalindromicSubstring.py]: It's good to know such things are possible but it'd be better to have two lines for better understanding, like:

```
L = max(L, l)
R = max(R, r + 1)
```

File: LongestPalindromicSubstring.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                r += 1
+                l -= 1
+            else:
+                l = r = (r + l) // 2 + 1  # going to midle +1 step
```
Comment (vitkarpov) [inline LongestPalindromicSubstring.py]: a typo

**Comment by vitkarpov** [comment]
Please, fill out the description using the following template:

### Problem

(copy & paste from leetcode)

### Approach

(describe your algorithm in plain English)

### Complexity

(big-O analysys)

File: LongestPalindromicSubstring.py (hunk 1 above)
Comment (Glazomer) [reply LongestPalindromicSubstring.py]: You are right, sorry

File: LongestPalindromicSubstring.py (hunk 3 above)
Comment (Glazomer) [reply LongestPalindromicSubstring.py]: Ye, not readable clearly. I made it better, wait for PR

File: LongestPalindromicSubstring.py (hunk 2 above)
Comment (Glazomer) [reply LongestPalindromicSubstring.py:12]: Logically it has not enough meaning. Just finding center moving to a function, idk( You think it's worth it?

Comment (vitkarpov) [reply LongestPalindromicSubstring.py:12]: It's up to you, I'd do that but it's not necessary for sure.

**Review by vitkarpov — APPROVED** [review]
Nice
